### ISSUE TITLE
@turf/kinks returns false self-intersections in MultiLineString features with branches 

### ISSUE THREAD

**Issue by ovdleondle** [Turfjs/turf#2756]
turf 7.1.0

When a MultiLineString feature has branches (e.g. a river with tributaries, or a street with side streets), kinks module returns self-intersection at the branching point. 
For this [geojson feature](https://geojson.io/#map=8.08/1/0.5):
`{"type":"FeatureCollection","features":[{"type":"Feature","properties":{},"geometry":{"type":"MultiLineString","coordinates":[[[0,0],[0,1],[0,2]],[[0,1],[1,1]]]}}]}`
![kinks](https://github.com/user-attachments/assets/0e540534-a345-4d45-b2f1-0957ccd2efff)
kinks would return a self-intersection at 0,1 coordinates, which is false.

**Comment by smallsaucepan** [comment]
Thanks for reporting this @ovdleondle.

Could it possibly be related to #2722? Would you mind checking to see if the behaviour is correct for v6.5.0?

**Comment by pm0u** [comment]
this appears to be unrelated to #2722:

- this module actually uses its own line-intersect algorithm
- this module, per its description in jsdoc, is intended to report _only_ self intersections

i am not sure that this qualifies as a bug per the documentation within turf but i could be missing some context for what would be expected of a module by this name. this case seems to be behaving as expected just by appearance

**Comment by smallsaucepan** [comment]
Thanks for ruling that out @pm0u. Will take a closer look at this one in due course then.

**Comment by ovdleondle** [comment]
@smallsaucepan I confirm that this behavior is not related to #2722, as the same issue occurs in version 6.5.0.

After reviewing the [kinks documentation](https://turfjs.org/docs/api/kinks), I also agree with @pm0u that the module is technically functioning as intended.

Our goal was to use the `kinks` module as a geometry validator on the client side. However, geometry validity is a nuanced concept, as described in the [PostGIS introduction](https://postgis.net/workshops/postgis-intro/validity.html):

> Validity is most important for polygons, which define bounded areas and require a good deal of structure. Lines are very simple and cannot be invalid, nor can points.
> Some of the rules of polygon validity feel obvious, and others feel arbitrary (and in fact, are arbitrary).

Additionally, geometry validation in [PostGIS](https://postgis.net/docs/en/using_postgis_dbmanagement.html#OGC_Validity) evaluates whether a geometry is simple and whether it is valid:

>  A simple geometry is one that has no anomalous geometric points, such as self intersection or self tangency.

> A MULTILINESTRING is simple only if all of its elements are simple and the only intersection between any two elements occurs at points that are on the boundaries of both elements.

> Geometry validity primarily applies to 2-dimensional geometries (POLYGONs and MULTIPOLYGONs) .

> For linear geometry the only validity rule is that LINESTRINGs must have at least two points and have non-zero length (or equivalently, have at least two distinct points.) Note that non-simple (self-intersecting) lines are valid.

To summarize, we plan to use the kinks module exclusively for polygon and multipolygon features. For lines, we will try using [lineIntersect](https://turfjs.org/docs/api/lineIntersect) with `ignoreSelfIntersections: false`.
